ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/RemoteModel.java

```diff
@@ -103,6 +103,10 @@ public void initModel(MLModel model, Map<String, Object> params, Encryptor encry
             Connector connector = model.getConnector().cloneConnector();
             connector
                 .decrypt(PREDICT.name(), (credential, tenantId) -> encryptor.decrypt(credential, model.getTenantId()), model.getTenantId());
+            // This situation can only happen for inline connector where we don't provide tenant id.
+            if (connector.getTenantId() == null && model.getTenantId() != null) {
```

Comment:
if model.getTenantId == null, this code won't do anything. But if that is an edge case to avoid in serverLess, you should check it and throw an exception.

Reply:
If both are null, that means this is a single tenant environment, so we don't need this anyway.